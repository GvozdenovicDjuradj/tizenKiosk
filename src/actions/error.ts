import { AnyAction } from "redux"
import { ERROR_SHOW, ERROR_RESET } from "./types"

export interface ShowError extends AnyAction {
  type: ERROR_SHOW;
  payload: string;
}

export interface ResetError extends AnyAction {
  type: ERROR_RESET
}

export type ErrorAction = ResetError | ShowError

export const showError = (payload: string): ShowError => ({
  type: ERROR_SHOW,
  payload
})

export const resetError = (): ResetError => ({ type: ERROR_RESET })
