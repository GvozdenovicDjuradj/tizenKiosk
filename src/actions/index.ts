export * from "./app"
export * from "./error"
export * from "./kiosk"
export * from "./kioskSettings"
export * from "./printing"
export * from "./navigationActions"
export * from "./screenSaver"
