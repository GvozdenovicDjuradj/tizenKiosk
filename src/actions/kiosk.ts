import { Action, AnyAction } from "redux"
import { APP, KIOSK, PRIVACY_POLICY_AGREE, PRIVACY_POLICY_DISAGREE } from "./types"
import {
  AddCustomerToQueueReset,
  AddCustomerToQueueSuccess,
  AddPrivateCustomerToQueueRequest,
  AddPrivateCustomerToQueueSuccess,
  AnyProduct,
  CustomerInQueue,
  DatePayload,
  ErrorPayload,
  FieldUpdatePayload,
  GetKioskDataSuccess,
  GetKioskSettingsSuccess,
  KioskSettings,
  Language,
  LanguagePayload,
  LoadPrivacyPolicySuccess,
  PrivacyPolicy,
  Product,
  ProductPayload,
  QuestionAnswerPayload,
  RegisterKioskFieldUpdate,
  RegisterKioskSuccess,
  StringPayload,
  NumberPayload
} from "../interfaces"
import { GET_KIOSK_ID_AND_DOMAIN_FROM_URL, SET_KIOSK_ID, SET_KIOSK_SERIAL, SET_KIOSK_URL } from "../constants/kiosk"

export const registerKiosk = (showPrint: boolean): AnyAction =>
  ({ type: KIOSK.REGISTER.REQUEST, hasPrinter: showPrint })

export const registerKioskFieldUpdate = (payload: FieldUpdatePayload): RegisterKioskFieldUpdate => ({
  payload,
  type: KIOSK.FIELD_UPDATE.REQUEST,
})

export const loadKioskSettings = (): Action => ({ type: KIOSK.SETTINGS.REQUEST })

export const loadScreenSaverData = (): Action => ({ type: KIOSK.LOAD_SCREEN_SAVER_DATA.REQUEST })

export const loadKioskQueueData = (): Action => ({ type: KIOSK.QUEUE_DATA.REQUEST })

interface KioskSettingsSuccessAction {
  type: string,
  payload: KioskSettings
}

export const kioskSettingsSuccess = (kioskSettings: KioskSettings): KioskSettingsSuccessAction => ({
  payload: kioskSettings,
  type: KIOSK.SETTINGS.SUCCESS
})

export const changeLanguage = (payload: Language): LanguagePayload => ({
  payload,
  type: KIOSK.SET_LANGUAGE.REQUEST,
})

export const setVenueCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.SET_COUNTRY.REQUEST
})

export const setGroupSize = (payload: number): NumberPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_GROUP_SIZE.REQUEST,
})

export const setCustomerName = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_NAME.REQUEST,
})

export const setCustomerEmail = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_EMAIL.REQUEST,
})

export const setCustomerCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_COUNTRY.REQUEST,
})

export const setCallingCode = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_CALLING_CODE.REQUEST,
})

export const setCustomerMobile = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_MOBILE.REQUEST,
})

export const setCustomerOrder = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_ORDER.REQUEST,
})

export const setNotes = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_NOTES.REQUEST,
})

export const setProduct = (payload: Product | AnyProduct): ProductPayload => ({
  payload,
  type: KIOSK.SET_PRODUCT.REQUEST,
})

export const setSubProduct = (payload: AnyProduct): ProductPayload => ({
  payload,
  type: KIOSK.SET_SUBPRODUCT.REQUEST,
})

export const addCustomerToQueue = (): Action => ({
  type: KIOSK.CUSTOMER.ADD_TO_QUEUE.REQUEST,
})

export const setAddCustomerToQueueError = (payload: string): StringPayload => ({
  type: KIOSK.CUSTOMER.ADD_TO_QUEUE.FAILURE,
  payload
})

export const setCheckInEmail = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_EMAIL.REQUEST,
  payload,
})

export const setCheckInMobile = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_MOBILE.REQUEST,
  payload,
})

export const setCheckInCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CHECK_IN.SET_COUNTRY,
})

export const setCheckInCallingCode = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CHECK_IN.SET_CALLING_CODE,
})

export const setCheckInOrder = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_ORDER.REQUEST,
  payload,
})

export const loadPrivacyPolicyDataSuccess = (payload: PrivacyPolicy, backgroundTask: boolean)
  : LoadPrivacyPolicySuccess => ({
    type: KIOSK.LOAD_PRIVACY_POLICY_DATA.SUCCESS,
    payload,
    backgroundTask
  })

export const addPrivateCustomerToQueueRequest = (queueId?: number, productId?: number)
  : AddPrivateCustomerToQueueRequest => ({
    type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.REQUEST,
    productId,
    queueId,
})

export const addPrivateCustomerToQueueError = (payload: ErrorPayload) => ({
  type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.FAILURE,
  payload
})

export const addPrivateCustomerToQueueSuccess = (customerInQueue: CustomerInQueue)
  : AddPrivateCustomerToQueueSuccess => ({
    type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.SUCCESS,
    payload: {
      customerInQueue
    }
  })

export const addCustomerToQueueSuccess = (customerInQueue: CustomerInQueue)
  : AddCustomerToQueueSuccess => ({
    type: KIOSK.CUSTOMER.ADD_TO_QUEUE.SUCCESS,
    payload: {
      customerInQueue
    }
  })

export const secretTap = () => ({
  type: APP.SECRET_TAP.REQUEST
})

export const addCustomerToQueueReset = (backgroundTask: boolean = false)
  : AddCustomerToQueueReset => ({
    type: KIOSK.CUSTOMER.ADD_TO_QUEUE.RESET,
    backgroundTask
  })

export const checkIn = (): Action => ({ type: KIOSK.CHECK_IN.REQUEST })

export const goToInitialScreen = (): Action => ({ type: APP.GO_INITIAL_SCREEN.REQUEST })

export const loadPrivacyPolicyDataRequest = (): Action => ({ type: KIOSK.LOAD_PRIVACY_POLICY_DATA.REQUEST })

export const agreePrivacyPolicy = (): Action => ({ type: PRIVACY_POLICY_AGREE })

export const disagreePrivacyPolicy = (): Action => ({ type: PRIVACY_POLICY_DISAGREE })

export const addPrivateCustomerToQueueFail = (): Action => ({ type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.FAILURE })

export interface GetKioskIdAndDomainFromUrlAction {
    type: GET_KIOSK_ID_AND_DOMAIN_FROM_URL
}

export const getKioskIdAndDomainFromUrl = (): GetKioskIdAndDomainFromUrlAction => ({
    type: GET_KIOSK_ID_AND_DOMAIN_FROM_URL
})

export interface SetKioskIdAction {
    type: SET_KIOSK_ID,
    payload: {
        id: string;
    }
}

export const setKioskId = (id: string): SetKioskIdAction => ({
    type: SET_KIOSK_ID,
    payload: {
        id
    }
})

export interface SetKioskSerialAction extends Action {
    type: SET_KIOSK_SERIAL,
    payload: {
        serial: string;
    }
}

export const setKioskSerial = (serial: string): SetKioskSerialAction => ({
    type: SET_KIOSK_SERIAL,
    payload: {
        serial
    }
})

export interface SetKioskUrlAction extends Action {
  type: SET_KIOSK_URL,
  payload: {
    url: string;
  }
}

export const setKioskUrl = (url: string): SetKioskUrlAction => ({
  type: SET_KIOSK_URL,
  payload: {
    url
  }
})

export type Actions =
  | Action
  | DatePayload
  | GetKioskDataSuccess
  | GetKioskSettingsSuccess
  | LanguagePayload
  | ProductPayload
  | QuestionAnswerPayload
  | RegisterKioskFieldUpdate
  | RegisterKioskSuccess
  | StringPayload
  | SetKioskUrlAction
  | SetKioskSerialAction
  | SetKioskIdAction
