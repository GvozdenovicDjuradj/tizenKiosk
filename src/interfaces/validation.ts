export interface ScreenFieldsState {
  [fieldName: string]: {
    valid: boolean;
    error: string[];
  },
}

export interface ValidationState {
  [screenName: string]: ScreenFieldsState
}

export interface ValidationChangeAction {
  type: string;
  payload: ValidationState;
}
