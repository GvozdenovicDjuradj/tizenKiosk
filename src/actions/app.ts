import { Action, APP, APPLICATION_START } from "./types"
import { AppScreens } from "../interfaces"

export const applicationStart = (): Action => ({ type: APPLICATION_START })

export const initApp =  (): Action => ({type: APP.START.REQUEST})

export const stopApp = (): Action => ({ type: APP.SHUTDOWN.REQUEST })

export const changeInitialPage = (payload: AppScreens): Action => ({
  type: APP.CHANGE_INITIAL_PAGE.REQUEST,
  payload,
})

export const getPermissionReadPhoneState = () => ({
  type: APP.PHONE_STATE_PERMISSION.REQUEST
})

export const deactivateApp = (): Action => ({ type: APP.RESET.REQUEST })
