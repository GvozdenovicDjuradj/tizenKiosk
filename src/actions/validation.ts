import { VALIDATION_STATE_CHANGE } from "./types"
import { ValidationChangeAction, ValidationState } from "../interfaces"

export const validationStateChange = (payload: ValidationState): ValidationChangeAction => ({
  type: VALIDATION_STATE_CHANGE,
  payload
})

export type Action = ValidationChangeAction
