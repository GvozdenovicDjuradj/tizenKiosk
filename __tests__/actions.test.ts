import configureMockStore, { MockStoreEnhanced } from "redux-mock-store"
import {
  addCustomerToQueue,
  changeInitialPage,
  changeLanguage,
  checkIn,
  checkNetwork,
  deactivateApp,
  goToInitialScreen,
  initApp,
  loadKioskQueueData,
  loadKioskSettings,
  registerKiosk,
  registerKioskFieldUpdate,
  saveKioskSettingsChanges,
  setCallingCode,
  setCheckInEmail,
  setCheckInMobile,
  setCheckInOrder,
  setCustomerCountry,
  setCustomerEmail,
  setCustomerMobile,
  setCustomerName,
  setCustomerOrder,
  setProduct,
  setSubProduct,
  setVenueCountry,
  stopApp,
  updateSettingsField,
  validateKioskSettings,
  secretTap,
  getPermissionReadPhoneState,
} from "../src/actions"
import { validationStateChange } from "../src/actions/validation"
import {
  AnyProduct,
  AppScreens,
  FieldUpdatePayload,
  Language,
  Product,
  RegisterKioskFields,
} from "../src/interfaces"

describe("Validation action test", () => {

  it(`Should return object with "type" and "payload"`, () => {
    const store = configureMockStore()({
      validation: {
        [AppScreens.HOME]: {
          url: {
            error: ["Some error message"],
            valid: false
          }
        }
      }
    })
    const change = {
      [AppScreens.HOME]: {
        url: { error: [], valid: true }
      }
    }
    expect(store.dispatch(validationStateChange(change))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

})

describe("Kiosk actions tests", () => {

  let store: MockStoreEnhanced

  beforeEach(() => {
    store = configureMockStore()()
  })

  it(`Should return object with property "type"`, () => {
    expect(store.dispatch(registerKiosk(true))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return object with property "type"`, () => {
    expect(store.dispatch(checkIn())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Customer Email (string) in payload`, () => {
    const payload = "test@qudini.com"
    expect(store.dispatch(setCheckInEmail(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Customer Email (string) in payload`, () => {
    const payload = "123456"
    expect(store.dispatch(setCheckInOrder(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Customer Mobile number (string) in payload`, () => {
    const payload = "+380663705600"
    expect(store.dispatch(setCheckInMobile(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return object with property "type"`, () => {
    expect(store.dispatch(addCustomerToQueue())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return object with payload for updating field value`, () => {
    const payload = {
      name: RegisterKioskFields.url,
      value: "https://qa.qudini.com"
    }
    expect(store.dispatch(registerKioskFieldUpdate(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Language as payload`, () => {
    const payload: Language = {
      countryCallingCode: "",
      countryIsoCode: "GB",
      countryName: "GB",
      languageIsoCode: "en",
      languageName: "en",
    }
    expect(store.dispatch(changeLanguage(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Customer country (string) in payload`, () => {
    const payload = "Country"
    expect(store.dispatch(setCustomerCountry(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with calling code (string) in payload`, () => {
    const payload = "+44"
    expect(store.dispatch(setCallingCode(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Customer Name (string) in payload`, () => {
    const payload = "Some name"
    expect(store.dispatch(setCustomerName(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Customer Mobile number (string) in payload`, () => {
    const payload = "+380663705600"
    expect(store.dispatch(setCustomerMobile(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Customer Email (string) in payload`, () => {
    const payload = "test@qudini.com"
    expect(store.dispatch(setCustomerEmail(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Customer Order number (string) in payload`, () => {
    const payload = "123456"
    expect(store.dispatch(setCustomerOrder(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action with Venue country (string) in payload`, () => {
    const payload = "UK"
    expect(store.dispatch(setVenueCountry(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return object with property "type"`, () => {
    expect(store.dispatch(goToInitialScreen())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return object with property "type"`, () => {
    expect(store.dispatch(loadKioskQueueData())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return object with property "type"`, () => {
    expect(store.dispatch(loadKioskSettings())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with Product in payload`, () => {
    const payload: Product = {
      waitTime: 0,
      id: 0,
      name: "test",
      queueId: 0,
      queueName: "test"
    }
    expect(store.dispatch(setProduct(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`Should return action with AnyProduct in payload`, () => {
    const payload: AnyProduct = {
      id: 0,
      name: "test",
      queueId: 0,
      queueName: "test",
      infoText: "info text",
      showInfo: true,
      products: [{
        id: "000",
        infoText: "info sub text",
        showInfo: true,
        subTitle: "subtitle",
      }]
    }
    expect(store.dispatch(setSubProduct(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

})

describe("Kiosk Settings actions tests", () => {

  let store: MockStoreEnhanced

  beforeEach(() => {
    store = configureMockStore()()
  })

  it(`should return action to trigger network check`, () => {
    expect(store.dispatch(checkNetwork())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action to trigger save kiosk settings`, () => {
    expect(store.dispatch(saveKioskSettingsChanges())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action to update kiosk settings field`, () => {
    const payload: FieldUpdatePayload = {
      name: RegisterKioskFields.kioskIdentifier,
      value: "test"
    }
    expect(store.dispatch(updateSettingsField(payload))).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action to trigger validation of kiosk settings
      and change Kiosk Identifier and/or Url if valid`, () => {
    expect(store.dispatch(validateKioskSettings())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

})

describe("App actions tests", () => {

  let store: MockStoreEnhanced

  beforeEach(() => {
    store = configureMockStore()()
  })

  it(`should return action to trigger App start`, () => {
    expect(store.dispatch(initApp())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action to trigger App stop`, () => {
    expect(store.dispatch(stopApp())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action to change initial App screen`, () => {
    const action = changeInitialPage(AppScreens.SERVICE_SELECTION)
    expect(store.dispatch(action)).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action to deactivate Kiosk`, () => {
    expect(store.dispatch(deactivateApp())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action to trigger secret tap`, () => {
    expect(store.dispatch(secretTap())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })

  it(`should return action to trigger get permission read phone state`, () => {
    expect(store.dispatch(getPermissionReadPhoneState())).toMatchSnapshot()
    expect(store.getActions()).toMatchSnapshot()
  })
})
