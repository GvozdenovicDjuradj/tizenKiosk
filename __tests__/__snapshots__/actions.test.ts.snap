// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App actions tests should return action to change initial App screen 1`] = `
Object {
  "payload": "SERVICE_SELECTION",
  "type": "CHANGE_INITIAL_PAGE_REQUEST",
}
`;

exports[`App actions tests should return action to change initial App screen 2`] = `
Array [
  Object {
    "payload": "SERVICE_SELECTION",
    "type": "CHANGE_INITIAL_PAGE_REQUEST",
  },
]
`;

exports[`App actions tests should return action to deactivate Kiosk 1`] = `
Object {
  "type": "APP_RESET_REQUEST",
}
`;

exports[`App actions tests should return action to deactivate Kiosk 2`] = `
Array [
  Object {
    "type": "APP_RESET_REQUEST",
  },
]
`;

exports[`App actions tests should return action to trigger App start 1`] = `
Object {
  "type": "APP_START_REQUEST",
}
`;

exports[`App actions tests should return action to trigger App start 2`] = `
Array [
  Object {
    "type": "APP_START_REQUEST",
  },
]
`;

exports[`App actions tests should return action to trigger App stop 1`] = `
Object {
  "type": "APP_SHUTDOWN_REQUEST",
}
`;

exports[`App actions tests should return action to trigger App stop 2`] = `
Array [
  Object {
    "type": "APP_SHUTDOWN_REQUEST",
  },
]
`;

exports[`App actions tests should return action to trigger get permission read phone state 1`] = `
Object {
  "type": "PHONE_STATE_PERMISSION_REQUEST",
}
`;

exports[`App actions tests should return action to trigger get permission read phone state 2`] = `
Array [
  Object {
    "type": "PHONE_STATE_PERMISSION_REQUEST",
  },
]
`;

exports[`App actions tests should return action to trigger secret tap 1`] = `
Object {
  "type": "APP_SECRET_TAP_REQUEST",
}
`;

exports[`App actions tests should return action to trigger secret tap 2`] = `
Array [
  Object {
    "type": "APP_SECRET_TAP_REQUEST",
  },
]
`;

exports[`Kiosk Settings actions tests should return action to trigger network check 1`] = `
Object {
  "type": "KIOSK_SETTINS_FORM_CHECK_NETWORK_REQUEST",
}
`;

exports[`Kiosk Settings actions tests should return action to trigger network check 2`] = `
Array [
  Object {
    "type": "KIOSK_SETTINS_FORM_CHECK_NETWORK_REQUEST",
  },
]
`;

exports[`Kiosk Settings actions tests should return action to trigger save kiosk settings 1`] = `
Object {
  "type": "KIOSK_SETTINS_FORM_SUBMIT_REQUEST",
}
`;

exports[`Kiosk Settings actions tests should return action to trigger save kiosk settings 2`] = `
Array [
  Object {
    "type": "KIOSK_SETTINS_FORM_SUBMIT_REQUEST",
  },
]
`;

exports[`Kiosk Settings actions tests should return action to trigger validation of kiosk settings
      and change Kiosk Identifier and/or Url if valid 1`] = `
Object {
  "type": "KIOSK_SETTINGS_FORM_VALIDATE_REQUEST",
}
`;

exports[`Kiosk Settings actions tests should return action to trigger validation of kiosk settings
      and change Kiosk Identifier and/or Url if valid 2`] = `
Array [
  Object {
    "type": "KIOSK_SETTINGS_FORM_VALIDATE_REQUEST",
  },
]
`;

exports[`Kiosk Settings actions tests should return action to update kiosk settings field 1`] = `
Object {
  "payload": Object {
    "name": "kioskIdentifier",
    "value": "test",
  },
  "type": "KIOSK_SETTINS_FORM_FIELD_UPDATE_REQUEST",
}
`;

exports[`Kiosk Settings actions tests should return action to update kiosk settings field 2`] = `
Array [
  Object {
    "payload": Object {
      "name": "kioskIdentifier",
      "value": "test",
    },
    "type": "KIOSK_SETTINS_FORM_FIELD_UPDATE_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with AnyProduct in payload 1`] = `
Object {
  "payload": Object {
    "id": 0,
    "infoText": "info text",
    "name": "test",
    "products": Array [
      Object {
        "id": "000",
        "infoText": "info sub text",
        "showInfo": true,
        "subTitle": "subtitle",
      },
    ],
    "queueId": 0,
    "queueName": "test",
    "showInfo": true,
  },
  "type": "KIOSK_SET_SUBPRODUCT_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with AnyProduct in payload 2`] = `
Array [
  Object {
    "payload": Object {
      "id": 0,
      "infoText": "info text",
      "name": "test",
      "products": Array [
        Object {
          "id": "000",
          "infoText": "info sub text",
          "showInfo": true,
          "subTitle": "subtitle",
        },
      ],
      "queueId": 0,
      "queueName": "test",
      "showInfo": true,
    },
    "type": "KIOSK_SET_SUBPRODUCT_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Customer Email (string) in payload 1`] = `
Object {
  "payload": "test@qudini.com",
  "type": "KIOSK_CHECK_IN_SET_EMAIL_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Customer Email (string) in payload 2`] = `
Array [
  Object {
    "payload": "test@qudini.com",
    "type": "KIOSK_CHECK_IN_SET_EMAIL_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Customer Email (string) in payload 3`] = `
Object {
  "payload": "123456",
  "type": "KIOSK_CHECK_IN_SET_ORDER_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Customer Email (string) in payload 4`] = `
Array [
  Object {
    "payload": "123456",
    "type": "KIOSK_CHECK_IN_SET_ORDER_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Customer Email (string) in payload 5`] = `
Object {
  "payload": "test@qudini.com",
  "type": "KIOSK_CUSTOMER_SET_EMAIL_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Customer Email (string) in payload 6`] = `
Array [
  Object {
    "payload": "test@qudini.com",
    "type": "KIOSK_CUSTOMER_SET_EMAIL_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Customer Mobile number (string) in payload 1`] = `
Object {
  "payload": "+380663705600",
  "type": "KIOSK_CHECK_IN_SET_MOBILE_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Customer Mobile number (string) in payload 2`] = `
Array [
  Object {
    "payload": "+380663705600",
    "type": "KIOSK_CHECK_IN_SET_MOBILE_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Customer Mobile number (string) in payload 3`] = `
Object {
  "payload": "+380663705600",
  "type": "KIOSK_CUSTOMER_SET_MOBILE_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Customer Mobile number (string) in payload 4`] = `
Array [
  Object {
    "payload": "+380663705600",
    "type": "KIOSK_CUSTOMER_SET_MOBILE_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Customer Name (string) in payload 1`] = `
Object {
  "payload": "Some name",
  "type": "KIOSK_CUSTOMER_SET_NAME_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Customer Name (string) in payload 2`] = `
Array [
  Object {
    "payload": "Some name",
    "type": "KIOSK_CUSTOMER_SET_NAME_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Customer Order number (string) in payload 1`] = `
Object {
  "payload": "123456",
  "type": "KIOSK_CUSTOMER_SET_ORDER_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Customer Order number (string) in payload 2`] = `
Array [
  Object {
    "payload": "123456",
    "type": "KIOSK_CUSTOMER_SET_ORDER_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Customer country (string) in payload 1`] = `
Object {
  "payload": "Country",
  "type": "KIOSK_CUSTOMER_SET_COUNTRY_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Customer country (string) in payload 2`] = `
Array [
  Object {
    "payload": "Country",
    "type": "KIOSK_CUSTOMER_SET_COUNTRY_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Language as payload 1`] = `
Object {
  "payload": Object {
    "countryCallingCode": "",
    "countryIsoCode": "GB",
    "countryName": "GB",
    "languageIsoCode": "en",
    "languageName": "en",
  },
  "type": "KIOSK_SET_LANGUAGE_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Language as payload 2`] = `
Array [
  Object {
    "payload": Object {
      "countryCallingCode": "",
      "countryIsoCode": "GB",
      "countryName": "GB",
      "languageIsoCode": "en",
      "languageName": "en",
    },
    "type": "KIOSK_SET_LANGUAGE_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with Product in payload 1`] = `
Object {
  "payload": Object {
    "id": 0,
    "name": "test",
    "queueId": 0,
    "queueName": "test",
    "waitTime": 0,
  },
  "type": "KIOSK_SET_PRODUCT_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with Product in payload 2`] = `
Array [
  Object {
    "payload": Object {
      "id": 0,
      "name": "test",
      "queueId": 0,
      "queueName": "test",
      "waitTime": 0,
    },
    "type": "KIOSK_SET_PRODUCT_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return action with calling code (string) in payload 1`] = `
Object {
  "payload": "+44",
  "type": "KIOSK_CUSTOMER_SET_CALLING_CODE_REQUEST",
}
`;

exports[`Kiosk actions tests Should return action with calling code (string) in payload 2`] = `
Array [
  Object {
    "payload": "+44",
    "type": "KIOSK_CUSTOMER_SET_CALLING_CODE_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return object with property "type" 1`] = `
Object {
  "hasPrinter": true,
  "type": "KIOSK_REGISTER_REQUEST",
}
`;

exports[`Kiosk actions tests Should return object with property "type" 2`] = `
Array [
  Object {
    "hasPrinter": true,
    "type": "KIOSK_REGISTER_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return object with property "type" 3`] = `
Object {
  "type": "KIOSK_CHECK_IN_REQUEST",
}
`;

exports[`Kiosk actions tests Should return object with property "type" 4`] = `
Array [
  Object {
    "type": "KIOSK_CHECK_IN_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return object with property "type" 5`] = `
Object {
  "type": "KIOSK_CUSTOMER_ADD_TO_QUEUE_REQUEST",
}
`;

exports[`Kiosk actions tests Should return object with property "type" 6`] = `
Array [
  Object {
    "type": "KIOSK_CUSTOMER_ADD_TO_QUEUE_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return object with property "type" 7`] = `
Object {
  "type": "APP_GO_INITIAL_SCREEN_REQUEST",
}
`;

exports[`Kiosk actions tests Should return object with property "type" 8`] = `
Array [
  Object {
    "type": "APP_GO_INITIAL_SCREEN_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return object with property "type" 9`] = `
Object {
  "type": "KIOSK_QUEUE_DATA_REQUEST",
}
`;

exports[`Kiosk actions tests Should return object with property "type" 10`] = `
Array [
  Object {
    "type": "KIOSK_QUEUE_DATA_REQUEST",
  },
]
`;

exports[`Kiosk actions tests Should return object with property "type" 11`] = `
Object {
  "type": "KIOSK_SETTINGS_REQUEST",
}
`;

exports[`Kiosk actions tests Should return object with property "type" 12`] = `
Array [
  Object {
    "type": "KIOSK_SETTINGS_REQUEST",
  },
]
`;

exports[`Kiosk actions tests should return action with Venue country (string) in payload 1`] = `
Object {
  "payload": "UK",
  "type": "KIOSK_SET_COUNTRY_REQUEST",
}
`;

exports[`Kiosk actions tests should return action with Venue country (string) in payload 2`] = `
Array [
  Object {
    "payload": "UK",
    "type": "KIOSK_SET_COUNTRY_REQUEST",
  },
]
`;

exports[`Kiosk actions tests should return object with payload for updating field value 1`] = `
Object {
  "payload": Object {
    "name": "url",
    "value": "https://qa.qudini.com",
  },
  "type": "KIOSK_FIELD_UPDATE_REQUEST",
}
`;

exports[`Kiosk actions tests should return object with payload for updating field value 2`] = `
Array [
  Object {
    "payload": Object {
      "name": "url",
      "value": "https://qa.qudini.com",
    },
    "type": "KIOSK_FIELD_UPDATE_REQUEST",
  },
]
`;

exports[`Validation action test Should return object with "type" and "payload" 1`] = `
Object {
  "payload": Object {
    "HOME": Object {
      "url": Object {
        "error": Array [],
        "valid": true,
      },
    },
  },
  "type": "VALIDATION_STATE_CHANGE",
}
`;

exports[`Validation action test Should return object with "type" and "payload" 2`] = `
Array [
  Object {
    "payload": Object {
      "HOME": Object {
        "url": Object {
          "error": Array [],
          "valid": true,
        },
      },
    },
    "type": "VALIDATION_STATE_CHANGE",
  },
]
`;
